docs/settings/python_analysis_ignore.md

```diff
@@ -0,0 +1,136 @@
+  - `?`: Matches a single character.
+
+### Use Cases
+
+- **Third-Party Libraries**: Suppress diagnostics in third-party libraries that might not adhere to strict type checking or have type stubs.
+- **Generated Code**: Ignore generated files that you don't edit manually and may contain code that triggers diagnostics.
+- **Legacy Code**: Exclude old code that you don't want to refactor immediately.
+
+## How to Use `python.analysis.ignore`
+
+You can configure `python.analysis.ignore` in your Visual Studio Code settings.
+
+### Setting in Visual Studio Code Settings
+
+To add `python.analysis.ignore` to your VS Code settings:
+
+1. **Open Settings (JSON)**:
+
+   - Click on the gear icon in the lower-left corner and select **Command Palette**.
+   - Type `Preferences: Open Settings (JSON)` and select it.
+
+2. **Add the Setting**:
+
+   - In your `settings.json`, add the `python.analysis.ignore` setting with the paths you want to ignore.
+
+   ```json
+   {
+       "python.analysis.ignore": ["**/legacy_code/**", "**/generated/**", "**/third_party/**"]
+   }
+   ```
+
+## Difference Between `ignore` and `exclude`
+
+While both `python.analysis.ignore` and [`python.analysis.exclude`](python_analysis_exclude.md) deal with controlling Pylance's behavior on certain files, they serve different purposes:
+
+- **`python.analysis.ignore`**:
+
+  - Suppresses diagnostics (errors and warnings) for the specified files or directories.
+  - The files are still processed if they are imported by other files, but no diagnostics are reported.
+  - If you open an ignored file, Pylance provides IntelliSense features but does not report errors or warnings.
+
+- **`python.analysis.exclude`**:
+
+  - Excludes the specified paths from the workspace.
+  - Pylance does not process these files for features like code navigation, auto-imports, or symbol searches.
+  - If an excluded file is imported by a non-excluded file, Pylance still reports diagnostics for the non-excluded file, potentially generating import errors.
+
+The `ignore` setting is useful when you want Pylance to recognize the files (e.g., for code completion when imported), but you don't want to see diagnostics for them.
+
+## Examples
+
+### Suppressing Diagnostics in Third-Party Libraries
+
+If you're using third-party libraries that are causing unwanted diagnostics, you can ignore them:
+
+```json
+{
+    "python.analysis.ignore": ["**/site-packages/**"]
+}
+```
+
+### Ignoring Generated Files
+
+Suppose your project generates code into a `generated` directory:
+
+```json
+{
+    "python.analysis.ignore": ["src/generated/**"]
+}
+```
+
+This way, Pylance won't show errors or warnings for the generated code but will still provide IntelliSense when you import from it.
+
+### Ignoring Files in Virtual Environments
+
+Sometimes, Pylance may display diagnostics for files in your virtual environment. You can suppress these by ignoring the virtual environment directories:
```

Comment:
Does this need to be explicitly set? If `exclude` is not specified by user, venv is ignored by default, right?

Reply:
ya, below, it talks about relationship between `exclude/ignore` but welcome any suggestions if you have better wording.

Reply:
Maybe add a clarification saying that if `exclude` is not set, the same default directories such - `**/node_modules`, `**/__pycache__`, `xxx`, directories are ignored by default.

Reply:
but it is not `ignored`, it is `excluded` in that case. 2 are different concept.

Reply:
I guess I can remove some of these examples since it is more confusing then clarifying things.

Reply:
I thought the `venv` is ignored by default, because I don't see any diagnostics in the error list generated for the files in `venv` directory.